rest/server_context_test.go

```diff
@@ -378,19 +376,23 @@ func TestTLSSkipVerifyCombinations(t *testing.T) {
 		t.Run(test.name, func(t *testing.T) {
 			startupConfig := &StartupConfig{
 				Bootstrap: BootstrapConfig{
+					Server:              base.UnitTestUrl(),
 					CACertPath:          test.caCert,
 					ServerTLSSkipVerify: test.serverTLSSkipVerify,
 				},
+				API: APIConfig{
+					HTTPS: HTTPSConfig{
+						UseTLSClient: base.BoolPtr(false),
+					},
+				},
 			}
 
-			sc, err := setupServerContext(startupConfig, false)
+			err := startupConfig.validate()
 			if test.expectError {
 				assert.Error(t, err)
 				assert.Contains(t, err.Error(), errorText)
-				assert.Empty(t, sc)
-			} else if err != nil {
-				// check if unrelated error
-				assert.NotContains(t, err.Error(), errorText)
```

Comment:
The point of this was to check that the error from the validation was not to do with TLS Skip Verify so then the validate function can have additional checks to the Startup Config in the future without having to change this test and without the test failing.

For example, this meant that the UseTLSClient config option didn't have to be added as it's errors would be ignored. 

assert.NotContains was the best way I could come up with to assert that the server TLS Skip verify error did not appear in the multiple error messages.

I would be interested in finding out if this is not a good practise to follow though.

Reply:
That's a fair comment. I think I can leave these checks as-is actually and have it all still work just fine.